scripts/scripts.js

```diff
@@ -132,15 +133,232 @@ export function decorateMain(main) {
+  if (elements.length > 0) {
+    fn();
+    elements.forEach((element) => {
+      if (element.style.visibility === 'hidden') {
+        element.style.visibility = 'visible';
+      }
+    });
+  }
+
+  const observer = new MutationObserver((mutations) => {
+    if (mutations.some((m) => m.target.tagName === 'BODY'
+        || m.target.dataset.sectionStatus === 'loaded'
+        || m.target.dataset.blockStatus === 'loaded')) {
+      fn();
+      mutations.forEach((m) => {
+        // Apply the desired operation on each mutation
+        if (m.target.style.visibility === 'hidden') {
+          m.target.style.visibility = 'visible';
+        }
+      });
+    }
+  });
+  // Watch sections and blocks being decorated async
+  observer.observe(document.querySelector('main'), {
+    subtree: true,
+    attributes: true,
+    attributeFilter: ['data-block-status', 'data-section-status'],
+  });
+  // Watch anything else added to the body
+  observer.observe(document.querySelector('body'), { childList: true });
+}
+
+function toCssSelector(selector) {
+  return selector.replace(/(\.\S+)?:eq\((\d+)\)/g, (_, clss, i) => `:nth-child(${Number(i) + 1}${clss ? ` of ${clss})` : ''}`);
+}
+
+async function getElementForProposition(proposition) {
+  const selector = proposition.data.prehidingSelector
+      || toCssSelector(proposition.data.selector);
+  console.log('selector:', selector);
+  return document.querySelector(selector);
+}
+
+function escapeSelector(selector) {
+  return selector.replaceAll(/#(\d)/g, '#\\3$1 ');
+}
+function getSectionByElementSelector(selector) {
+  try {
+    let section = document.querySelector(escapeSelector(selector));
+    while (section && !section.classList.contains('section')) {
+      section = section.parentNode;
+    }
+    return section;
+  } catch (error) {
+    console.error('Error occurred while querying selector:', error);
+    return null; // Return null if an error occurs
+  }
+}
+
+async function getAndApplyRenderDecisions() {
+  // Get the decisions, but don't render them automatically
+  // so we can hook up into the AEM EDS page load sequence
+  const response = await window.alloy('sendEvent', { renderDecisions: false });
+  const { propositions } = response;
+
+  propositions.forEach((entry) => {
+    // Iterate over the array of items in each entry
+    if (entry.items) {
+      entry.items.forEach((item) => {
+        if (item.schema === 'https://ns.adobe.com/personalization/dom-action') {
+          const cssSelector = item.data.selector;
+          const { prehidingSelector } = item.data;
+          let section;
+          if (prehidingSelector) {
+            section = getSectionByElementSelector(prehidingSelector);
+          }
+          if (!section && cssSelector) {
+            section = getSectionByElementSelector(cssSelector);
+          }
+          if (section) {
+            section.style.visibility = 'hidden';
```

Comment:
Can we do this via application of a CSS class like `hidden` ?

Reply:
@sdmcraft done